cli/progress-bar/src/terminal.rs

```diff
@@ -129,10 +129,13 @@ impl TerminalContext<'_> {
             return Guard { sender: None };
         };
 
-        self.steps_left = self
-            .steps_left
-            .checked_sub(1)
-            .expect("steps number overflow");
+        self.steps_left = match self.steps_left.checked_sub(1) {
+            Some(n) => n,
+            None => {
+                tracing::warn!("Progress bar steps number overflow, resetting to 0");
```

Comment:
Calling `display_step` more than declared `num_steps` times is a logical error that should to panic. Moreover, superconsole renders the whole terminal and this log will be lost.